### ISSUE TITLE
Clarify that only one interestedParty statement reference should be supplied

### ISSUE THREAD

**Issue by Blueskies00** [openownership/data-standard#467]
It is possible when generating json to fill in all of the different interestedParty values for a single ownership or control statement.

The schema browser says the following: 

> The interested party has some level of ownership or control over the entity referenced in this ownership or control statement. This should be described with reference to either an entity statement or person statement, or, where the interested party is unknown, details of why.

The description does include "or", however this is easy to overlook and create json with a combination or all of these. While creating/updating tooling to reflect the mutual exclusiveness will be important, it may be of benefit to highlight more explicitly that only one of these values should be completed.

The data validation tool also doesn't pick this up, so I've raised an [issue](https://github.com/openownership/lib-cove-bods/issues/91) there too.

I will review the BODS Generator workbook to see if I can mitigate the issue further there.

**Comment by lgs85** [comment]
This is a good point. In the longer term, we could think about addressing this by changing how we model `subject` and `interestedParty`. So instead of the current model:

```json
   "interestedParty": {
      "describedByPersonStatement": "019a93f1-e470-42e9-957b-03559861b2e2"
    }
```
We could consider something like:

```json
   "interestedParty": {
      "statementID": "019a93f1-e470-42e9-957b-03559861b2e2",
      "statementType":"personStatement"
    }
```

The latter is less efficient, but has advantages of being clearer, and will produce nicer outputs from flattening tools.

**Comment by kd-ods** [comment]
And to bring @lgs85 ' suggestion in line with the current proposal on [core vs. metadata](https://docs.google.com/spreadsheets/d/1-X077bxagedLBH7DijncMQmXMTPe0_w1IaiyJlui3bg/edit#gid=0) restructuring in statements:

```json
   "interestedParty": {
      "recordID": "019a93f1-e470-42e9-957b",
      "recordType":"Person"
    }
```

Where recordType can be Person, Entity or None; and recordID must be specified if the recordType is Person or Entity.

**Comment by kd-ods** [comment]
Schema-wise in 0.4, both `interestedParty` and `subject` could have a recordReference object as their value:

```
   "interestedParty": {
      "recordID": "019a93f1-e470-42e9-957b",
      "recordType":"Person"
    }
```
For `subject` the `recordType` would always be "Entity"

**Comment by rhiaro** [comment]
@kd-ods and I talked about this today. I wondered if this could be simplified by defining the value of `interestedParty` to be a record ID string only, ie:

```
"interestedParty": "019a93f1-e470-42e9-957b"
```

Pros: 

* it would be the same structure for `interestedParty` as for `subject`
* the information about the record type is not duplicated from the record itself (reducing moving parts and sources of truth)

Cons:

* the record type information in this place is a useful hint for anyone processing data and then looking up the record details eg. from an external API (where the action might be different for people vs for companies)

I'm in favour of simplifying, consistency between the two properties, and reducing bytes. However, if there is a concrete use case where there's a strong need for the extra nesting and type data, we should keep the nested structure. It would be interesting to know if/how people are using the type data in `interestedParty` at the moment, or plan to in the future.

**Comment by kd-ods** [comment]
@rhiaro  - there's a complication: we need a way of handling non-identification of an interested party. The [current schema has this](https://standard.openownership.org/en/0.3.0/schema/reference.html#interestedparty), which is a bit clunky. Can we discuss in the next couple of days? I'm doing some docs work which depends on figuring this out.

**Comment by rhiaro** [comment]
In this case, if I understand correctly, we could:

1. omit the `interestedParty` field altogether
2. set the value of the `interestedParty` field to `null`
3. set the value of the `interestedParty` field to a reserved string, like `unspecified`
4. set the value to self-reference the statementID of the Relationship Statement it's a part of, as a sort of pointer to where more information can be found (even though if you're processing this field you're already there).

In each case, the rest of the information about why the interested part is unknown would still be contained within the Relationship Statement, alongside (or instead of) this field, right?

1. has the disadvantage that it could be done by accident, rather than intentionally, and validating which it is is impossible
2. and 3. are probably the easiest to to reason about when parsing the data, as long as the docs are clear. 3. might be a bit easier for a human to understand, and also serves as a pointer to the `unspecified` _field_ which contains the reasoning (assuming we're keeping that). But validating based on a specific string value compared to any other string value (if the value is an ID) feels a bit more hacky than if we were doing it by datatype (as in 2).

4 . is most likely a terrible idea.

**Comment by kd-ods** [comment]
Yes - I think 5 is best. And we can use the current ['Unspecified or unknown ownership and control' object](https://github.com/openownership/data-standard/blob/05854dbac374fc77adfe2807e8d61d7b020823a1/schema/ownership-or-control-statement.json#L106).

**Comment by kd-ods** [comment]
Assigning to @rhiaro since we talked about rolling this change in with the initial schema redesign.

**Comment by kd-ods** [comment]
Closed via PR #598
